gensim/models/keyedvectors.py

```diff
@@ -337,15 +337,18 @@ def __getitem__(self, key_or_keys):
 
         return vstack([self.get_vector(key) for key in key_or_keys])
 
-    def get_index(self, key):
+    def get_index(self, key, default=None):
         """Return the integer index (slot/position) where the given key's vector is stored in the
         backing vectors array.
 
         """
-        if key in self.key_to_index:
-            return self.key_to_index[key]
-        elif isinstance(key, (int, np.integer)) and key < len(self.index_to_key):
+        val = self.key_to_index.get(key, -1)
+        if val >= 0:
+            return val
+        elif isinstance(key, (int, np.integer)) and key < len(self.index_to_key) and key >= 0:
```

Comment:
Watch out – `isinstance(True, int)` holds too (not sure if that's a problem here).

```suggestion
        elif isinstance(key, (int, np.integer)) and 0 <= key < len(self.index_to_key):
```

Reply:
The int-ness of `True` shouldn't become an issue - if someone tries to pass that instead of the expected key types, they've earned whatever error or undefined behavior they get.